packages/verified-fetch/src/verified-fetch.ts

```diff
@@ -328,25 +358,71 @@ export class VerifiedFetch {
       }
     }
 
+    /**
+     * range-request validation. This shouldn't be required, but unixfs.cat isn't throwing in some cases.
+     */
+    if (isRangeRequest) {
+      /**
+       * if requested CID was a directory, we already have a unixfs stat object
+       * if requested CID was a file or raw, we need to fetch the stat object
+       */
+      stat = stat ?? await this.unixfs.stat(resolvedCID, {
+        signal: options?.signal,
+        onProgress: options?.onProgress
+      })
+
+      if (suffixLength != null) {
+        const rangeWithSize = getRequestRange(options?.headers, stat.fileSize)
+        if (rangeWithSize != null) {
+          offset = rangeWithSize.offset
+          length = rangeWithSize.length
+        }
+      }
+      const resp = this.checkForInvalidRangeRequest({ stat, offset, length, resource })
+      if (resp != null) {
+        return resp
+      }
+    }
+
     const asyncIter = this.unixfs.cat(resolvedCID, {
       signal: options?.signal,
-      onProgress: options?.onProgress
+      onProgress: options?.onProgress,
+      offset,
+      length
     })
     this.log('got async iterator for %c/%s', cid, path)
 
-    const { stream, firstChunk } = await getStreamFromAsyncIterable(asyncIter, path ?? '', this.helia.logger, {
-      onProgress: options?.onProgress
-    })
-    const response = okResponse(resource, stream, {
-      redirected
-    })
-    await this.setContentType(firstChunk, path, response)
+    try {
+      const { stream, firstChunk } = await getStreamFromAsyncIterable(asyncIter, path ?? '', this.helia.logger, {
+        onProgress: options?.onProgress
+      })
+      let response: Response
+      if (isRangeRequest) {
+        // type assertion says stat may still be undefined, but we've set it again in the above check for isRangeRequest
+        const totalSize = Number(stat?.fileSize)
+        offset = offset ?? 0
+        response = okRangeResponse(resource, stream, { offset, length, totalSize }, {
+          redirected
+        })
+      } else {
+        response = okResponse(resource, stream, {
+          redirected
+        })
+      }
+      await this.setContentType(firstChunk, path, response)
 
-    if (ipfsRoots != null) {
-      response.headers.set('X-Ipfs-Roots', ipfsRoots.map(cid => cid.toV1().toString()).join(',')) // https://specs.ipfs.tech/http-gateways/path-gateway/#x-ipfs-roots-response-header
-    }
+      if (ipfsRoots != null) {
+        response.headers.set('X-Ipfs-Roots', ipfsRoots.map(cid => cid.toV1().toString()).join(',')) // https://specs.ipfs.tech/http-gateways/path-gateway/#x-ipfs-roots-response-header
+      }
 
-    return response
+      return response
+    } catch (err) {
+      this.log.error('Error streaming %c/%s', cid, path, err)
```

Comment:
Just a tiny consistency thing, all of our errors are thrown with messages in sentence case, logging is all done in lower case.  Not sure why, but it's where we are.

```suggestion
      this.log.error('error streaming %c/%s', cid, path, err)
```

Reply:
fixed in [`f357a3d` (#10)](https://github.com/ipfs/helia-verified-fetch/pull/10/commits/f357a3d5eaa240a4808c8461cce65619fba2f1e4)